lib/wakes/location_helper.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+class LocationHelper
```

Comment:
@benhutton yes, only used in the matcher. It allows the spec to continue using strings to represent locations—preserving readability—but it extends it to also support domain names (as you probably already deduced).

For name, I had also considered `LocationString` but didn't love either name. Do you like that better? It's a lightweight wrapper, halfway between a string and a Wakes location. 

I'll take a look at wrapping `URI()`. I wanted to keep it lightweight since it's just for the spec, but my approach is definitely fragile.